spark/src/main/resources/python/zeppelin_pyspark.py

```diff
@@ -365,10 +345,20 @@ def getCompletion(self, text_value):
           mod = ast.Module([node])
           code = compile(mod, '<stdin>', 'exec')
           exec(code, _zcUserQueryNameSpace)
+
+        intp.setStatementsFinished("", False)
       except:
-        raise Exception(traceback.format_exc())
+        exception = traceback.format_exc()
+        m = re.search("File \"<stdin>\", line (\d+).*", exception)
+        if m:
+          line_no = int(m.group(1))
+          intp.setStatementsFinished(
+            "Fail to execute line {}: {}\n".format(line_no, stmts[line_no - 1]) + exception, True)
+        else:
+          intp.setStatementsFinished(exception, True)
+      else:
+        intp.setStatementsFinished("", False)
 
-    intp.setStatementsFinished("", False)
   except Py4JJavaError:
```

Comment:
does this still trap Py4jJavaError correctly? looks like this will never be hit?